lib/stream.js

```diff
@@ -15,8 +15,15 @@ var log = isABib ? bunyan.createLogger({name: 'sierra-bib-retriever'}) : bunyan.
 // send data to kinesis Stream
 exports.streamPoster = (records, streamName) => {
   return writeToStreamsClient(streamName, records)
+          .then(() => {
+            return Promise.resolve({sent: records.length, receieved: records.length})
```

Comment:
That's right, @rhernand3z. The expectation is for the recursive function to make sure to resolve only when there are no failed records. So, when streamPoster() gets that response it knows everything has been sent correctly.